src/sliding_sync.rs

```diff
@@ -940,23 +957,71 @@ async fn async_main_loop(
 
     let (client, _sync_token) = match new_login_opt {
         Some(new_login) => new_login,
-        None => loop {
-            log!("Waiting for login request...");
-            if let Some(login_request) = login_receiver.recv().await {
-                log!("Received login request for user {}", login_request.user_id);
-                match login(Cli::from(login_request)).await {
-                    Ok(new_login) => break new_login,
-                    Err(e) => {
-                        error!("Login failed: {e:?}");
-                        Cx::post_action(LoginAction::LoginFailure(e.to_string()));
-                        enqueue_rooms_list_update(RoomsListUpdate::Status {
-                            status: e.to_string(),
-                        });
+        None => {
+            let server_name = ServerName::parse("matrix.org")?;
+            let unauth_client = Client::builder().server_name(&server_name).build().await?;
+            let login_type_res = unauth_client.matrix_auth().get_login_types().await?;
+            let Some(LoginType::Sso(sso_type)) = login_type_res.flows.get(0) else { return Err(anyhow!("Login Type response error")); };
+            Cx::post_action(LoginAction::IdentityProvider(sso_type.identity_providers.clone()));
+            loop {
+                log!("Waiting for login request...");
+                if let Some(login_type) = login_receiver.recv().await {
+                    match login_type {
+                        LoginRequest::LoginByPassword(login_request) =>{
+                            log!("Received login request for user {}", login_request.user_id);
```

Comment:
now that we have multiple kind of login options, let's be precise in our language.
```suggestion
                            log!("Received password-based login request for user {}", login_request.user_id);
```